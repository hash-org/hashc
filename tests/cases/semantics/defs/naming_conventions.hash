// run=fail, stage=semantic
Boo := struct(
    arg,
    max: i32,
)

Moo := struct(
    arg,
    max,
)

Foo := struct(
    bax: i32,
    max: i32,
)

Mux := enum(
    Crux(
        bax: i32,
        max: i32,
    ),
    Tux(
        bax,
        max,
    ),
    Lux(
        bax: i32,
        max,
    ),
)
