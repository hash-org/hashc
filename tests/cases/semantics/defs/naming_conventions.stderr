error: mismatching naming convention of fields within a struct
 --> $DIR/naming_conventions.hash:4:5
3 |       arg,
4 |       max: i32,
  |       ^^^^^^^^ this is expected to be a un-named field
5 |   );
  = note: fields of a {origin} should all be named, or un-named since mixing naming conventions leads to ambiguities
